Feature/tc 08

Haris-moin opened the pull request.

hmoin-nisum-com committed:
TC-08: Design Tweet Card

- create layout of tweet card
- create tweet.json file with actual tweet data
- use json data in card layout
- create card design without icone

hmoin-nisum-com committed:
TC-08: Design Tweet Card

- synced with develop
- resolved conflicts

hmoin-nisum-com committed:
TC-08: Design Tweet Card

- create some common components for code reusability
- create icons from style components
- create global icons file
- create tweet card with feedback icons

hmoin-nisum-com committed:
TC-08: Design Tweet Card

- complete tweet card design
- use followers with user data
- add light blue color on theme
- set as different color on hashtag and mentions as well ass get there data on click by a function
- set and modified tweet card design

hiqbalmallick commented on src/App.tsx:
```diff
@@ -1,4 +1,8 @@
-import Counter from "./example/counter";
+import "./App.css";
```
No change is required on App.tsx. You just need to create the component and that's it.

hiqbalmallick commented on src/components/common/Title/index.tsx:
```diff
@@ -0,0 +1,17 @@
+import React from "react";
+import "./style.css";
+import { VerifiedIcon } from "../../../icons/styled-icons";
```
why aren't we using styled-components?

hmoin-nisum-com committed:
TC-08-Design Tweet Card

- remove css files and create styled components
- create mock directory for mock data

hiqbalmallick commented on src/App.tsx:
```diff
@@ -1,17 +1,18 @@
         <MainLayout />
         <div className="App">
-          <Counter />
+          <Counter />å
```
please remove this typo

hmoin-nisum-com committed:
TC-08 Design Tweet Card

- remove type code in app.tsx

hmoin-nisum-com committed:
TC-08 Design Tweet Card

- Restructured code
- create interface directory

hiqbalmallick commented on src/__mock__/data.json:
```diff
@@ -0,0 +1,96 @@
+{
```
update name of this file to tweets.json

hiqbalmallick commented on src/components/common/avatar/index.tsx:
```diff
@@ -0,0 +1,15 @@
+import React from "react";
+import { Avatar, ImgContainer } from "./style";
+
+const UserAvatar: React.FC<{
+  url: string;
+  style: { width: string; height: string };
+}> = ({ url, style }) => {
```
replace this with interfaces

hiqbalmallick replied:
this one is not done, why this was marked as resolved?

hiqbalmallick commented on src/interfaces/index.ts:
```diff
@@ -0,0 +1,3 @@
+export * from "./components/avatar/intex";
```
please fix this typo here and also in actual file name

hmoin-nisum-com committed:
TC-08 Design tweet card

- correct file name
- remove inline typing by interface

hmoin-nisum-com committed:
TC-08 Design Tweet Card

- create duration method for created at

hiqbalmallick commented on src/components/tweet-card/index.tsx:
```diff
@@ -0,0 +1,111 @@
+      ),
+    };
+  };
+  useEffect(() => {
+    hashTag();
+    getCreatedAt();
+  });
```
why does this useEffect is invoking these two methods? that two without the dependency array?

Haris-moin replied:
ohh sorry by mistake i removed its dependency.

hiqbalmallick commented on src/components/tweet-card/index.tsx:
```diff
@@ -0,0 +1,111 @@
+import moment from "moment";
+
+const TweetCard: React.FC = () => {
+  const [createdAt, setCreatedAt] = useState("");
```
unnecessary state. we should utilize the prop value directly for date transformation.

hiqbalmallick commented on src/components/tweet-card/index.tsx:
```diff
@@ -0,0 +1,111 @@
+      console.log(el.textContent);
+    }
+  };
+  const getCreatedAt = () => {
+    const duration = moment.duration(
+      moment(new Date()).diff(moment(tweet.data.created_at))
+    );
+    const hours = +duration.asHours();
+    if (hours < 0) {
+      setCreatedAt(String(hours % 60));
+    } else if (hours > 24) {
+      setCreatedAt(moment(tweet.data.created_at).format("MMMM DD"));
+    } else setCreatedAt(String(hours));
+  };
```
this method is transforming date into a particular format, please make it a pure function rather than using state.

hiqbalmallick commented on src/components/tweet-card/index.tsx:
```diff
@@ -0,0 +1,111 @@
+import UserName from "../common/user-name";
+import moment from "moment";
+
+const TweetCard: React.FC = () => {
```
expect prop for Tweet object instead of mock data, mock data was used to get the object info only. Once a component has been created, it should expect a prop.

Haris-moin replied:
ok

hmoin-nisum-com committed:
TC-08: Design Tweet Card

- create tweet and user interface
- create helper file for pure functions
- remove tweet json and pass data as props
- resolved remains issues

hmoin-nisum-com committed:
removed un used imports

hmoin-nisum-com committed:
Revert "removed un used imports"

This reverts commit 0d3ac1bc442208c95c205ea6d8bcec9e3efcd9c2.

hmoin-nisum-com committed:
TC-08: Design Tweet Card

- create tweet and user interface
- create helper file for pure functions
- remove tweet json and pass data as props
- resolved remains issues

hmoin-nisum-com committed:
Merge branch 'feature/TC-08' of github.com:AnasShahid/TwitterCrawler into feature/TC-08

hmoin-nisum-com committed:
TC-08: Design Tweet Card

- create tweet and user interface
- create helper file for pure functions
- remove tweet json and pass data as props
- resolved remains issues

hmoin-nisum-com committed:
Merge branch 'feature/TC-08' of github.com:AnasShahid/TwitterCrawler into feature/TC-08
